apache-nifi.yaml

```diff
@@ -46,43 +51,86 @@ pipeline:
 
+      # Create expected directories
       for dir in conf database_repository flowfile_repository content_repository provenance_repository state; do
-        mkdir -p ${{targets.destdir}}/usr/share/nifi/logs/${dir}
+        mkdir -p "${{targets.contextdir}}/${NIFI_HOME}/${dir}"
       done
 
-  - runs: |
-      echo "#!/bin/sh\n" > ${{targets.destdir}}/usr/share/nifi/bin/nifi-env.sh
+      # Set shell in environment script
+      echo "#!/bin/sh\n" > "${{targets.contextdir}}/${NIFI_HOME}/bin/nifi-env.sh"
 
-  - runs: |
-      mkdir -p ${{targets.destdir}}/usr/share/nifi/scripts
-      install -m755 nifi-docker/dockerhub/sh/*.sh "${{targets.destdir}}"/usr/share/nifi/scripts/
-      chmod -R +x ${{targets.destdir}}/usr/share/nifi/scripts
-      rm ${{targets.destdir}}/usr/share/nifi/bin/*.bat
+      # Install scripts
+      mkdir -p "${{targets.contextdir}}/${NIFI_BASE}/scripts"
+      install -m755 nifi-docker/dockerhub/sh/*.sh "${{targets.contextdir}}/${NIFI_BASE}/scripts/"
+
+      # Ensure scripts are executable
+      chmod -R +x "${{targets.contextdir}}/${NIFI_BASE}/scripts"
+
+      # Fix paths in scripts
+      sed -i "s|/opt|/usr/share|g" "${{targets.contextdir}}/${NIFI_BASE}"/scripts/*.sh
+
+      # Remove batch scripts
+      rm "${{targets.contextdir}}/${NIFI_HOME}"/bin/*.bat
+
+      # Symlink current release to versioned path
+      ln -sf "${NIFI_HOME}" "${{targets.contextdir}}/${NIFI_BASE}/nifi-${{package.version}}"
 
   - uses: strip
 
 subpackages:
+  - name: "${{package.name}}-toolkit"
+    description: "Toolkit for Apache NiFi"
+    dependencies:
+      runtime:
+        - apache-nifi
+    pipeline:
+      - runs: |
+          TARGET_DIR="nifi-toolkit/nifi-toolkit-assembly/target"
+
+          # Unzip toolkit
+          unzip "${TARGET_DIR}/nifi-toolkit-1.26.0-bin.zip" -d "${TARGET_DIR}"
+
+          # Install toolkit
+          mkdir -p "${{targets.contextdir}}/${NIFI_BASE}"
+          cp -ar "${TARGET_DIR}/nifi-toolkit-${{package.version}}" "${{targets.contextdir}}/${TOOLKIT_HOME}"
+
+          # Remove batch scripts
+          rm "${{targets.contextdir}}/${TOOLKIT_HOME}"/bin/*.bat
+
+          # Symlink current release to versioned path
+          ln -sf "${TOOLKIT_HOME}" "${{targets.contextdir}}/${NIFI_BASE}/nifi-toolkit-${{package.version}}"
+      - uses: strip
+    test:
+      pipeline:
+        - runs: |
+            echo "help" | /usr/share/nifi/nifi-toolkit-current/bin/cli.sh
+
   - name: "${{package.name}}-compat"
     description: "Compatibility package to place binaries in the location expected by upstream image"
+    dependencies:
+      runtime:
+        - apache-nifi
     pipeline:
       - runs: |
-          mkdir -p "${{targets.contextdir}}"/${NIFI_HOME}
-          ln -sf /usr/share/nifi "${{targets.contextdir}}"/${NIFI_HOME}/nifi
+          mkdir -p "${{targets.contextdir}}/opt"
+          ln -sf "${NIFI_BASE}" "${{targets.contextdir}}/opt/nifi"
```

Comment:
we may need 3 more symlinks something like:

```
ln -sf /usr/share/nifi-toolkit "${{targets.contextdir}}"/${NIFI_TOOLKIT_HOME}
ln -sf /${NIFI_HOME} /opt/nifi/nifi-${{package.version}}
ln -sf /${NIFI_TOOLKIT_HOME} /opt/nifi/nifi-toolkit-${{package.version}}
```

`$NIFI_BASE` is set to `/usr/share/` above.

* https://github.com/apache/nifi/blob/338083f6850b61cd2651e41bbd73b3cb5330d98c/nifi-docker/dockerhub/Dockerfile#L63
* https://github.com/apache/nifi/blob/338083f6850b61cd2651e41bbd73b3cb5330d98c/nifi-docker/dockerhub/Dockerfile#L78

Reply:
We can get away with one. NIFI_BASE is set to /usr/share/nifi. Confirmed all paths are correct afterwards with NiFi and NiFi toolkit installed